var/spack/repos/builtin/packages/muparser/package.py

```diff
@@ -22,21 +22,31 @@ class Muparser(Package):
     # https://github.com/beltoforion/muparser/pull/46
     patch("auto_ptr.patch", when="@2.2.5")
 
-    depends_on("cmake@3.1.0:", when="@2.2.6:", type="build")
+    variant("samples", default=True, description="enable samples", when="build_system=cmake")
+    variant("openmp", default=True, description="enable OpenMP support", when="build_system=cmake")
+    variant(
+        "wide_char",
+        default=False,
+        description="enable wide character strings in place of ASCII",
+        when="build_system=cmake",
+    )
+    variant("shared", default=True, description="enable shared libs", when="build_system=cmake")
 
-    # Cmake build since 2.2.6
-    @when("@2.2.6:")
-    def install(self, spec, prefix):
-        cmake_args = ["-DENABLE_SAMPLES=OFF", "-DENABLE_OPENMP=OFF", "-DBUILD_SHARED_LIBS=ON"]
+    depends_on("cmake@3.1.0:", when="@2.2.6:", type="build")
```

Comment:
I just recalled that inheriting from `CMakePackage` will add an automatic dep like

```
depends_on(cmake, when=build_system=cmake)
```

so this should not be necessary. Other than that this looks good